### PR TITLE
Playback 2024: Longest Episode

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: #2250 |
|:---:|

Part of #2346

| | |
| -- | -- |
| <video src="https://github.com/user-attachments/assets/2d8b04f7-fa50-4fbd-b183-908d5b88b1e3"> | <video src="https://github.com/user-attachments/assets/95c4d00d-14b9-4463-8780-173bd528619b"> |

## To test

* Enable the `endOfYear2024` feature flag
* Navigate through the stories until you hit this Longest Episode screen
* Ensure layout looks alright across devices

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/End of Year/Stories/2024/LongestEpisode2024Story.swift [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    Image("playback-sticker-phew")
+                        .resizable()
+                        .frame(width: stickerSize.width, height: stickerSize.height)
+                        .position(x: -6, y: 0, for: stickerSize, in: geometry.frame(in: .global), corner: .topTrailing)
+                }
+                .frame(width: geometry.size.width * 0.9)
+                .padding(.top, isSmallScreen ? 0 : 20)
+                VStack(alignment: .leading, spacing: isSmallScreen ? 4 : 16) {
+                    let timeString = episode.playedUpTo.storyTimeDescriptionForSharing
+                    Text("The longest episode you listened to was \(timeString)")
```
Comment (SergioEstevao) [inline podcasts/End of Year/Stories/2024/LongestEpisode2024Story.swift]: Is the EOY English only? No need to localize these?

File: podcasts/End of Year/Stories/2024/LongestEpisode2024Story.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    let podcast: Podcast
+
+    @State var firstCover: Double = 0.4
+    @State var secondCover: Double = 0.32
+    @State var thirdCover: Double = 0.24
+    @State var fourthCover: Double = 0.16
+    @State var fifthCover: Double = 0.08
+    @State var sixthCover: Double = 0
+
+    var backgroundColor = Color(hex: "#E0EFAD")
```
Comment (SergioEstevao) [inline podcasts/End of Year/Stories/2024/LongestEpisode2024Story.swift]: Does this color ever change? Can it be a let variable?

**Review by SergioEstevao — APPROVED** [review]
Tested on iPhone 6s and iPad and it looks correct, iPhone 6s maybe a bit tight.
| iPad | iPhone 6s| 
| - | - | 
| Simulator Screenshot - iPad (10th generation) - 2024-10-29 at 15 36 32 |  Simulator Screenshot - iPhone 6s - 2024-10-29 at 15 28 01 |



Regarding the code there is just some nitpick comments.

**Pull request merged**